images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,349 @@
+       returned: success
+       type: complex
+"""
+
+import copy
+import traceback
+import datetime
+import kubernetes.config.dateutil
+import openshift
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+# user args
+EVENT_ARG_SPEC = {
+    "state": {"default": "present", "choices": ["present", "absent"],},
+    "name": {"required": True},
+    "namespace": {"required": True},
+    "merge_type": {"type": "list", "choices": ["json", "merge", "strategic-merge"]},
+    "message": {"type": "str", "required": True},
+    "reason": {"type": "str", "required": True},
+    "reportingComponent": {"type": "str", "required": True},
+    "type": {"choices": ["Normal", "Warning"],},
+    "source": {"type": "str", "component": {"type": "str", "required": True}},
+    "involvedObject": {
+        "type": "dict",
+        "apiVersion": {"type": "str", "required": True},
+        "kind": {"type": "str", "required": True},
+        "name": {"type": "str", "required": True},
+        "namespace": {"type": "str", "required": True},
+    },
+}
+
+
+class KubernetesEvent(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubernetesEvent, self).__init__(*args, k8s_kind="Event", **kwargs)
+
+    @property
+    def argspec(self):
+        """ argspec property builder """
+        argumentSpec = copy.deepcopy(AUTH_ARG_SPEC)
+        argumentSpec.update(EVENT_ARG_SPEC)
+        return argumentSpec
+
+    def execute_module(self):
+        """ Module execution """
+        #auth
+        self.client = self.get_api_client()
+
+        # retrieve params
+        message = self.params.get("message")
+        reason = self.params.get("reason")
+        reportingComponent = self.params.get("reportingComponent")
+        eventType = self.params.get("type")
+        source = self.params.get("source")
+
+        definition = defaultdict(defaultdict)
+
+        meta = definition["metadata"]
+        meta["name"] = self.params.get("name")
+        meta["namespace"] = self.params.get("namespace")
+
+        involvedObj = self.params.get("involvedObject")
+        print(involvedObj["kind"],involvedObj["namespace"],involvedObj["name"],involvedObj["apiVersion"])
+
+
+        resource = self.find_resource(
+            "Event", "v1", fail=True)
+
+        priorEvent = None
+        try:
+            priorEvent = resource.get(
+                name=meta["name"],
+                namespace=meta["namespace"])
+
+        except openshift.dynamic.exceptions.NotFoundError:
+            pass
+
+        # event counter
+        priorCount = 1
+        print("the count is no event exists %i" % priorCount)
+        now = datetime.datetime.now()
+        rfc = kubernetes.config.dateutil.format_rfc3339(now)
+        firstTimestamp = rfc
+        lastTimestamp = rfc
+
+        if priorEvent:
+            if priorEvent["reason"] != reason:
```

Comment:
This entire branch is redundant now since this case is just setting the values to the same as what they're initialized to. We can just change this to handle only when reason is the same since the initial values handle when it's different and we don't need redundantly set them.

Eg:
```
        # default values are the same values to use as when the reason changes.
        priorCount = 1
        now = datetime.datetime.now()
        rfc = kubernetes.config.dateutil.format_rfc3339(now)
        firstTimestamp = rfc
        lastTimestamp = rfc

        # reason didn't change
        if priorEvent and priorEvent["reason"] == reason:
            priorCount = priorEvent["count"] + 1
            firstTimestamp = priorEvent["firstTimestamp"]
            lastTimestamp = rfc
```